test/BumpVersionSpec.hs

```diff
@@ -53,70 +59,118 @@ setOverrides overrides = do
 
+randomSemVer :: Gen SemVer
+randomSemVer = SemVer <$> arbitrary <*> arbitrary <*> arbitrary <*> pure [] <*> pure []
+
 spec :: Spec
-spec = around_ setup $ do
+spec = describe "spago bump-version" $ do
+  describe "property tests" $ do
+
+    describe "getNextVersion" $ do
+
+      prop "Major version bump should +1 major version and set minor and patch to 0" $
+        forAll randomSemVer $ \initialVersion ->
+            let Right newVersion = getNextVersion Major initialVersion
+            in   _svMajor newVersion == _svMajor initialVersion + 1
+              && _svMinor newVersion == 0
+              && _svPatch newVersion == 0
+
+      prop "Minor version bump should keep major, +1 minor and set patch to 0" $
+        forAll randomSemVer $ \initialVersion ->
+            let Right newVersion = getNextVersion Minor initialVersion
+            in   _svMajor newVersion == _svMajor initialVersion
+              && _svMinor newVersion == _svMinor initialVersion + 1
+              && _svPatch newVersion == 0
+
+      prop "Patch version bump should keep major and minor and +1 patch" $
+        forAll randomSemVer $ \initialVersion ->
+            let Right newVersion = getNextVersion Patch initialVersion
+            in   _svMajor newVersion == _svMajor initialVersion
+              && _svMinor newVersion == _svMinor initialVersion
+              && _svPatch newVersion == _svPatch initialVersion + 1
+
+      prop "parseVersion . unparseVersion == id" $
+        forAll randomSemVer $ \v -> parseVersion (unparseVersion v) == Right v
+
+  describe "unit tests" $ do
+
+    describe "parseVersionBump" $ do
+
+      it "should parse 'major'" $
+        parseVersionBump "major" `shouldBe` Just Major
+
+      it "should parse 'minor'" $
+        parseVersionBump "minor" `shouldBe` Just Minor
 
-  describe "spago bump-version" $ do
+      it "should parse 'patch'" $
+        parseVersionBump "patch" `shouldBe` Just Patch
 
-    it "Spago should complain when no git repo exists" $ do
+      it "should parse version starting with 'v'" $
+        parseVersionBump "v1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-      spago ["bump-version", "minor"] >>= shouldBeFailureInfix
-        "Your git working tree is dirty. Please commit or stash your changes first"
+      it "should parse version not starting with 'v'" $
+        parseVersionBump "1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-    before_ (initGitTag "v1.2.3") $ it "Spago should not make a tag when not passing --no-dry-run" $ do
+      -- TODO is this desired behavior, or should we just drop ONE 'v'? I'd agree it's edge case, but still :-)
+      it "should drop multiple 'v's from the beginning" $
+        parseVersionBump "vvvvvvvv1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-      spago ["bump-version", "minor"] >>= shouldBeSuccess
-      getHighestTag >>= (`shouldBe` Just "v1.2.3")
+      -- TODO should this work or should we strip these in parser implementation?
```

Comment:
PREREL / META should probably be allowed - for one thing because there are existing packages that have such versions - e.g. `halogen` `v5.0.0-rc.7`.